app/models/goal.py

```diff
@@ -1,5 +1,27 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column, relationship
 from ..db import db
+from typing import List, Optional
+
+from typing import TYPE_CHECKING
+if TYPE_CHECKING: from .task import Task
 
 class Goal(db.Model):
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str]
+    tasks: Mapped[List["Task"]] = relationship("Task", back_populates="goal")  
+    
+    @classmethod
+    def from_dict(cls, goal_data):
+        new_goal = Goal(title=goal_data["title"])
+        return new_goal
+
+    def to_dict(self, include_tasks=False, include_task_ids=False):
+        goal_dict = {
+            "id": self.id,
+            "title": self.title
+        }
+        if include_tasks:
+            goal_dict["tasks"] = [task.to_dict() for task in self.tasks]
+        if include_task_ids:  
+            goal_dict["task_ids"] = [task.id for task in self.tasks]
```

Comment:
We can simplify this instance method because we don't need to pass any arguments to it. 

self.tasks will be a list with elements or an empty list so we can use this to check if we should include tasks for a goal. 

Also, per the [project documentation for Wave 6](https://github.com/Ada-C23/task-list-api/blob/main/ada-project-docs/wave_06.md#sending-a-list-of-task-ids-to-a-goal), the requirement is that the response body for a POST request to /goals/id/tasks should be:
```py
{
  "id": 1,
  "task_ids": [1, 2, 3]
}
```

Therefore, your `to_dict` function does not need to create a dictionary with key `tasks` and value is a list of dictionaries representing tasks (what you wrote on lines 23-24) so this should be removed. 

```suggestion
    def to_dict(self):
        goal_dict = {
            "id": self.id,
            "title": self.title
        }
        if include_task_ids:  
            goal_dict["task_ids"] = [task.id for task in self.tasks]
```